### DISCUSSION TITLE
Extension settings support

### DISCUSSION THREAD

**Discussion by SleepingShell** [zed-industries/zed#17226] [Feature Requests]
Hi, I was wondering if extensions (not a language extension) can have their own settings that the user configures in the `settings.json` file? This would be useful for configuring the extension. Specifically, I am working on a Slash Command.

Separately, I was wondering if there is any information on the proper way to call async functions (such as a suggested blocking method) from the extension trait?

**Comment by widersky** [comment]
Maybe we'll see these and other important functions when Zed supports all possible LLM's in the world 🤷‍♂️

**Comment by ItsHarper** [comment]
This already exists. The Java extension README documents how to configure it, for example: https://github.com/zed-extensions/java

Edit: I didn't realize you specified non-language extensions specifically. Here's the list of the types of extensions you can provide settings for (which, yeah, seems limited to MCP servers and languages): https://docs.rs/zed_extension_api/latest/zed_extension_api/settings/index.html

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/community-links
